src/graph/utils/pm/nested_matcher.cpp

```diff
@@ -252,7 +254,8 @@ bool match_node_inputs(const binding_t &b, match_context_t *ctx,
     if (node_inputs_matcher.get_node()->get_inputs().size()
             == VARIADIC_INPUT_NUM) {
         matching_status = node_inputs_matcher.match_variadic_inputs();
-    } else if (!has_commutative_inputs(node_inputs_matcher.get_op())) {
+    } else if (!has_commutative_inputs(node_inputs_matcher.get_op(),
+                       {b.bind_op_port, b.bind_port})) {
```

Comment:
I'm wondering, for select, whose first input is non-commutative, and second & third inputs are commutative, how can we make sure all the inputs are correctly visited and matched?

Reply:
Take https://github.com/intel-sandbox/onednn-graph-perf-analysis/blob/master/json/mha/bf16/mha-pt-starcoder-inf-bf16-bs1.json as an example. When `bind_node` is `select`, `bind_port` is 2 and `bind_op_port` is 1 which conform to `commutative inputs = {1, 2}`.

Reply:
I understand that. I mean how is the 0-th input matched? As it's non-commutative.

Reply:
Do you mean the first or the second input will swap with the first one? I doubt whether that's a valid graph, since the data type of the 0-th input is boolean after all. As for other cases, like when all three input data types are the same but only two are commutative, I don't think we have such a requirement.